packages/parse5-html-rewriting-stream/test/rewriting-stream.test.ts

```diff
@@ -305,4 +305,48 @@ describe('RewritingStream', () => {
 
         assert.throws(() => stream.write(buf), TypeError);
     });
+
+    it('Should pass long text correctly (GH-292)', (done) => {
+        const source = 'a'.repeat((1 << 16) + 1);
+        const parser = new RewritingStream();
+        let output = '';
+
+        parser.on('data', (data) => {
+            output += data.toString();
+        });
+
+        parser.once('close', () => {
+            try {
+                assert.strictEqual(output.length, source.length);
+                done();
+            } catch (error) {
+                done(error);
+            }
+        });
+
+        parser.write(source);
+        parser.end();
+    });
+
+    it.only('Should emit comment after text correctly', (done) => {
+        const source = `${'a'.repeat((1 << 16) - 3)}<!-- foo -->`;
+        const parser = new RewritingStream();
+        let output = '';
+
+        parser.on('data', (data) => {
+            output += data.toString();
+        });
+
+        parser.once('close', () => {
```

Comment:
Switching this line to the `finish` event will make the test fail, due to the stream buffer waterline being hit. Other tests have been updated to also use the `close` event, for consistency.